### PR TITLE
Support image formats AVIF, HEIC and JXL

### PR DESCRIPTION
**Pull request by ausi**
Once the next version of https://github.com/php-imagine/Imagine gets released this pull request should add support for AVIF, HEIC and JXL.

I didn’t add them to the default value of `contao.image.valid_extensions` as there is no GD support yet.

### CODE REVIEW

File: core-bundle/src/Resources/contao/config/mimetypes.php [hunk 1]
```diff
@@ -103,6 +103,9 @@
 	'svg'   => array('image/svg+xml', 'iconSVG.svg'),
 	'svgz'  => array('image/svg+xml', 'iconSVGZ.svg'),
 	'webp'  => array('image/webp', 'iconWEBP.svg'),
+	'avif'  => array('image/avif', 'iconAVIF.svg'),
+	'heic'  => array('image/heic', 'iconHEIC.svg'),
+	'jxl'  => array('image/jxl', 'iconJXL.svg'),
```
Comment (ausi) [inline core-bundle/src/Resources/contao/config/mimetypes.php:108]: @leofeyer who usually generates these icons?
The webp icon was added in https://github.com/contao-components/contao/commit/3a003eca5b9d9dfdfc10f5b0ae965d3bdddd39df?short_path=d22bb60#diff-d22bb60ec705ee366f3a5f37c2d77143cd2fc712d58b0d518ed1b7e7d2618fbd

Comment (leofeyer) [reply core-bundle/src/Resources/contao/config/mimetypes.php:108]: I will provide them in the next days.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/config/mimetypes.php:108]: Done. See https://github.com/contao-components/contao/commit/44abd0b24e5508ff427b7e722ddf6f8617591fc8.

File: core-bundle/src/Resources/contao/library/Contao/File.php [hunk 2]
```diff
@@ -269,6 +269,9 @@ public function __get($strKey)
 									'jpeg' => IMAGETYPE_JPEG,
 									'png' => IMAGETYPE_PNG,
 									'webp' => IMAGETYPE_WEBP,
+									'avif' => \defined('IMAGETYPE_AVIF') ? IMAGETYPE_AVIF : 19,
+									'heic' => IMAGETYPE_UNKNOWN, // TODO: replace with IMAGETYPE_HEIC once available
+									'jxl' => IMAGETYPE_UNKNOWN, // TODO: replace with IMAGETYPE_JXL once available
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/File.php:274]: When will these constants be available?

Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/File.php:274]: When (and if at all) HEIC and JXL are supported in GD and PHP.

Related: https://github.com/libgd/libgd/issues/699 and https://github.com/libgd/libgd/issues/414

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/File.php:274]: So HEIC and JXL might never be supported? Good that we have already added the icons. 😄

Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/File.php:274]: Not supported by GD. But Imagick already supports all of them.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/File.php:274]: Ah, I see. 👍

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**